### ISSUE TITLE
Gluon, Minpy and NN

### ISSUE THREAD

**Issue by feiyulv** [apache/incubator-mxnet#7286]
It seems all those three packages are developed to suppport dynamical graph, which can provide flexible programming interface as pytorch.  Will Minpy and NN be discarded?  And will mxnet continue to complete Gluon without moving to develop a new package?  I 'm a little confused by these packages.

**Comment by jermainewang** [comment]
Thanks for the question. For MinPy project, our initial design goals include xx parts: (1) dynamic autograd; (2) fully-compatible numpy support; (3) transparent integration with MXNet's symbolic programming so one-style-achieves-all. During the process, we learned three things:

* We initially intend to release MinPy as a separate package, but it seems that dynamic autograd is so popular (look at pytorch) that MXNet requires this feature very urgently. Therefore, we decide to merge the autograd feature into MXNet (see https://github.com/apache/incubator-mxnet/blob/master/python/mxnet/contrib/autograd.py) by @ZihengJiang .
* A good DNN model builder and an easy-to-use toolchain (initializer, updater, etc.) is crucial for fast development. MinPy's NN module is based on CS231n, which is not sophisticated enough for researches. Therefore, we spend a lot of efforts on a better model builder structure ( @piiswrong @sneakerkg and many others). The new package is developed directly in MXNet (also called NN) and now becomes Gluon.
* Speed is a dominating factor for a DL framework. We found writing fine-grained operators in python yields many efficiency issues. Therefore, the MinPy team is focusing on Just-In-Time (JIT) compilation to improve the efficiency (please stay tuned!).

**TL;DR**: Gluon (successor of NN package) will be our standard DNN model & solver package. MinPy will evolve more into two directions: (1) JIT functionality that can also benefit Gluon and other MXNet packages; (2) More compatible support for NumPy operators.

@zzhang-cn

**Comment by feiyulv** [comment]
@jermainewang thx for your explanation
